clients/src/main/java/org/apache/kafka/common/record/FileLogInputStream.java

```diff
@@ -56,7 +57,7 @@ public FileChannelLogEntry nextEntry() throws IOException {
             return null;
 
         logHeaderBuffer.rewind();
-        channel.read(logHeaderBuffer, position);
+        Utils.readFully(channel, logHeaderBuffer, position);
```

Comment:
As Ismael said, with the patch, it seems that there is no need to do the logHeaderBuffer.hasRemaining() check in the next line? Ditto in a few other places.